weaviate/collections/classes/internal.py

```diff
@@ -563,3 +590,28 @@ def from_input(
             include_vector=include_vector,
             is_group_by=group_by is not None,
         )
+
+
+_GenerativeReturn = Union[_GenerativeQueryReturn[P, R], _GenerativeGroupByReturn[P, R]]
+
+# The way in which generic typealiases work requires that all the generic arguments
+# are listed first and in the order of their apperance in the typealias.
+# GenerativeReturn[Properties, References, TProperties, TReferences] is the intended use and so
+# these four generics appear first. All others resolve afterwards correctly
+GenerativeReturn = Union[
+    _GenerativeReturn[Properties, References],
+    _GenerativeReturn[TProperties, TReferences],
+    _GenerativeReturn[Properties, CrossReferences],
+    _GenerativeReturn[Properties, TReferences],
+    _GenerativeReturn[TProperties, References],
+    _GenerativeReturn[TProperties, CrossReferences],
+]
+
+GenerativeQueryReturn = Union[
```

Comment:
Currently, `group_by` is not supported in `bm25`, `hybrid`, and `fetch_objects` so their methods can only return one of `_GenerativeQueryReturn`. Rather than have the whole `Union[...]` type defined in the method, I've defined `GenerativeQueryReturn` to take its place instead

If and when `group_by` is implemented for these types of queries, this type will change to `GenerativeReturn`

Reply:
thanks!